pyproject.toml

```diff
@@ -7,43 +7,133 @@ license = "MIT"
+    'Intended Audience :: Developers',
+    'License :: OSI Approved :: MIT License',
+    'Operating System :: OS Independent',
+    'Programming Language :: Python',
+    'Programming Language :: Python :: 3',
+    'Programming Language :: Python :: 3.7',
+    'Programming Language :: Python :: 3.8',
+    'Programming Language :: Python :: 3.9',
+    'Programming Language :: Python :: 3.10',
+    'Programming Language :: Python :: 3.11',
+    'Topic :: Documentation',
+    'Topic :: Utilities',
+    'Framework :: Sphinx',
+]
 
 [tool.poetry.dependencies]
-python = "^3.7.0"
+python = ">=3.8,<4"
 click = "^8.0.3"
-rich = "^11.2.0"
 Jinja2 = "^3.0.3"
 memray = "^1.3.1"
-sphinx = "^5.3.0"
 pyinstrument = "^4.3.0"
+rich = "^11.2.0"
 snakeviz = "^2.1.1"
+sphinx = "^5.3.0"
 
-[tool.poetry.dev-dependencies]
+[tool.poetry.group.dev.dependencies]
+black = "^23.7.0"
+ruff = "^0.0.285"
+
+[tool.poetry.group.docs.dependencies]
+# some shared dependencies as the app itself is called in the docs build
+click = "^8.0.3"
+memray = "^1.3.1"
+pyinstrument = "^4.3.0"
+rich = "^11.2.0"
+sphinx = "^5.3.0"
+sphinxcontrib-programoutput = "^0.17"
 
 [tool.poetry.scripts]
-sphinx-performance = 'sphinx_performance.performance:cli_performance'
 sphinx-analysis = 'sphinx_performance.analysis:cli_analysis'
+sphinx-performance = 'sphinx_performance.performance:cli_performance'
+
+[tool.ruff]
+select = ["ALL"] # Enable all checks and maintain an ignore list
+# a lot of ignores here due to activation of ruff linting in an established project
+# candidates for reactivation are marked with a star
+ignore = [
+    "ANN001",  # missing-type-function-argument *
+    "ANN003",  # missing-type-kwargs *
+    "ANN101",  # missing-type-self
+    "ANN201",  # missing-return-type-undocumented-public-function *
+    "ANN202",  # missing-return-type-private-function *
+    "B905",    # zip-without-explicit-strict *
+    "C901",    # complex-structure *
+    "D100",    # undocumented-public-module *
+    "D101",    # undocumented-public-class *
+    "D102",    # undocumented-public-method *
+    "D107",    # no docstring needed for __init__
+    "D203",    # D203 and D211 are mutually exclusive: https://github.com/PyCQA/pydocstyle/issues/141
+    "D212",    # D212 and D213 are mutually exclusive: https://stackoverflow.com/a/45990465
+    "FBT002",  # boolean-default-value-positional-argument *
+    "PERF401", # manual-list-comprehension *
+    "PLR0912", # too-many-branches *
+    "PLR0913", # too-many-arguments *
+    "PLR0915", # too-many-statements *
+    "PLW0603", # global-statement *
+    "PLW1510", # subprocess-run-without-check *
+    "S603",    # subprocess-without-shell-equals-true *
+    "S607",    # start-process-with-partial-path *
+]
+fixable = ["ALL"]
+
+# Allow autofix for all enabled rules (when `--fix`) is provided.
+# fixable = ["A", "B", "C", "D", "E", "F", "..."]
+# unfixable = []
+
+# Exclude a variety of commonly ignored directories.
+exclude = [
```

Comment:
is this really necessary? ruff already respects the gitignore https://beta.ruff.rs/docs/settings/#respect-gitignore

Reply:
Good point, I removed the list